packages/plugin-ext/src/plugin/types-impl.ts

```diff
@@ -1754,6 +1754,55 @@ export class WorkspaceEdit implements theia.WorkspaceEdit {
     }
 }
 
+export class DataTransferItem {
+    asString(): Thenable<string> {
+        return Promise.resolve(typeof this.value === 'string' ? this.value : JSON.stringify(this.value));
+    }
+
+    asFile(): theia.DataTransferFile | undefined {
+        return undefined;
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    constructor(readonly value: any) {
+    }
+}
+
+/**
+ * A map containing a mapping of the mime type of the corresponding transferred data.
+ *
+ * Drag and drop controllers that implement {@link TreeDragAndDropController.handleDrag `handleDrag`} can add additional mime types to the
+ * data transfer. These additional mime types will only be included in the `handleDrop` when the the drag was initiated from
+ * an element in the same drag and drop controller.
+ */
+@es5ClassCompat
+export class DataTransfer implements Iterable<[mimeType: string, item: DataTransferItem]> {
```

Comment:
Hi, is there a specific reason why the constructor which exists in vscode for this class is not is not implemented here?

Reply:
also why is the list implemented as `Map<string, DataTransferItem> `when in vscode its `Map<string, DataTransferItem[]>`

Reply:
[`vscode.d.ts`](https://github.com/microsoft/vscode/blob/main/src/vscode-dts/vscode.d.ts)  does not specify a constructor. Neither does it mandate any particular implementation.

Reply:
thats true. Though while i'm currently working on #12025 i saw that i probably will need to convert from a VSDataTransfer object to our implementation which is implemented with Map<string, IDataTransferItem[]>

Reply:
im just not totally sure what the Impact of that could be. But i would guess it could get more difficult to support drag and dop of multiple files

Reply:
Wouldn't those be dragged as a `uri-list` anyway? Native DnD does not support multiple items per type, see https://html.spec.whatwg.org/multipage/dnd.html or https://developer.mozilla.org/en-US/docs/Web/API/DataTransfer/setData. That said: it's a good test case: what happens when dragging multiple file of the same type into the test view?

Reply:
Anyway: since the _interface_ of `DataTransferItem` only supports a single item per MIME type, I don't really see the utility of having a differing implementation.